### PR TITLE
chore: refactor top-level package, add utilities, bugfixes

### PR DESCRIPTION
**Pull request by kzantow**
This PR adjusts fangs such that:
* everything moves to the top-level, e.g. `github.com/anchore/fangs` instead of the subdir of `github.com/anchore/fangs/config`
* `yaml` tags are used by default, avoiding the need to duplicate with yet more `mapstructure` tags
  * one caveat: for embedded structs to be squashed, they must include the nonstandard `,squash` option in the `yaml` tag
* adds a `Summarize` function that, given a struct (presumably with default or otherwise set values) will output a nicely formatted yaml document including default values, descriptions, and env vars
  * includes descriptions from: `description` struct tags, cobra `Command` objects, directly specified descriptions
  * the latter two must be manually configured (with simple utilities: `NewCommandDescriber` and `NewDescribers()`
* adds a `SummarizeLocations` function that given a config returns all the locations known based on the provided `FinderDescriber`s
* corrects some miscellaneous documentation issues

### CODE REVIEW

File: summarize.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type Describer interface {
```
Comment (wagoodman) [inline summarize.go:13]: nit: what about `FieldDescriber` with a function `DescribeFields`?

File: finders.go
Comment (wagoodman) [inline finders.go:1]: should we change a finder to take `fs.FS` to help with effects? (does not need to be addressed in this PR)

File: finders.go [hunk 2]
```diff
@@ -22,22 +21,12 @@ func FindDirect(cfg Config) []string {
...
-func FindConfigYamlInCwd(cfg Config) []string {
-	// check if config.yaml exists in the current directory
-	f := "./config.yaml"
-	if fileExists(f) {
-		cfg.Logger.Warnf("DEPRECATED: %s as a configuration file is deprecated and will be removed as an option in v1.0.0, please rename to .syft.yaml", f)
-		return []string{f}
-	}
-
-	return nil
+func FindConfigYamlInCwd(_ Config) []string {
```
Comment (wagoodman) [inline finders.go:28]: should we even include this finder at all in this lib?

Comment (kzantow) [reply finders.go:28]: I think it's a possibly valid option someone would want -- to load `./config.yaml`, which is why I've included it here (even though it's a "bug"). It's nice to be already tested consistently, this way.

**Review by wagoodman — APPROVED** [review]

File: finders.go
Comment (kzantow) [reply finders.go:1]: Apparently Viper has `SetFs(fs afero.Fs)`; so we'd need to use that interface...

Comment (kzantow) [reply finders.go:1]: Agree this could be addressed later.

**Pull request merged**